server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -200,8 +199,6 @@ private static ValuesSourceType getLegacyMapping(
             return CoreValuesSourceType.GEOPOINT;
         } else if (fieldType instanceof RangeFieldMapper.RangeFieldType) {
             return CoreValuesSourceType.RANGE;
-        } else if (indexFieldData instanceof IndexHistogramFieldData) {
-            return CoreValuesSourceType.HISTOGRAM;
```

Comment:
Correct.  Definitely not for multi/array/top_metrics, and composite _shouldn't_ be able to support it because that would only be possible with a `percentiles` agg inside composite which will resolve separately (either OSS and fail or XPack and succeed).

But... I'm gonna add some tests to verify that. Just in case :)

Reply:
Couldn't find a way to actually test this, since Core can't use the histo field without the xpack plugin, and testing from xpack doesn't make sense since the field and agg are available.  

But I think that confirms that this is ok to remove

The only other user of histo is `boxplot` which is xpack only too.